pkg/kubernetes/provider.go

```diff
@@ -116,10 +124,23 @@ func (p *ProviderStore) GenerateProviderNodeAffinity(provider string) []corev1.N
 	return nsrList
 }
 
-// generateProviderName creates a provider name from the cloud provider and provider value
-// this should not be used to create a resource name as it may contain underscores
+// generateProviderName creates a provider name from the cloud provider and
+// provider value. NOTE: this should not be used to create a resource name as
+// it may contain underscores
 func generateProviderName(cloudProvider, providerValue string) string {
-	return cloudProvider + "-" + providerValue
+	if isProviderValueAllowed(providerValue) {
```

Comment:
could we either rename `generateProviderName` , or move this check outside the function? as it stands, it doesn't seem intuitive for validation to be occurring here as well